sdk/core/azure-core-amqp/src/models/amqp_value.cpp

```diff
@@ -671,9 +738,32 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
     return os;
   }
 
+  std::ostream& operator<<(std::ostream& os, AmqpArray const& value)
+  {
+    // Let the AmqpValue specialization handle serialization of the array.
+    AmqpValue arrayValue(value);
+    os << arrayValue;
+    return os;
+  }
+
+  std::ostream& operator<<(std::ostream& os, AmqpMap const& value)
+  {
+    // Let the AmqpValue specialization handle serialization of the map.
+    AmqpValue mapValue(value);
+    os << mapValue;
+    return os;
+  }
+  std::ostream& operator<<(std::ostream& os, AmqpSymbol const& value)
+  {
+    // Let the AmqpValue specialization handle serialization of the array.
+    AmqpValue arrayValue(value);
+    os << arrayValue;
+    return os;
+  }
+
   bool AmqpValue::IsNull() const
```

Comment:
should there also be ==/!= operators for `(AmqpValue const& val, std::nullptr_t)` + `(std::nullptr_t, AmqpValue const& val)` calling `val.IsNull()`?

That may allow semantics like `if (amqpValue == nullptr)`, if we want that.